lib/datadog/core/configuration/option.rb

```diff
@@ -94,6 +96,87 @@ def default_precedence?
 
         private
 
+        def coerce_env_variable(value)
+          case @definition.type
+          when :int
+            value.to_i
```

Comment:
Will this mask empty strings as `0`?
```ruby
''.to_i
# => 0
```

I think this is `to_i` here because `env_to_int`'s current implementation is `to_i`. This is fair, as we don't want to break compatibility, despite this implementation not being strictly correct.

If we need to keep `to_i` here to ensure backwards compatibility, can I ask you to document it here (and also around `#to_f` below) that we want to make this more strict in ddtrace 2.0? All invalid values becoming zero is definitely not what we want here.

Reply:
Definitely worth adding a `# TODO:` comment explaining that.

Reply:
Done https://github.com/DataDog/dd-trace-rb/pull/2970/commits/15246fc9b9ecd88deeeff161e9fd3e8bf2b7cc5e